apps/prairielearn/src/pages/administratorCourseRequests/administratorCourseRequests.js

```diff
@@ -30,62 +29,9 @@ router.post(
     if (!res.locals.is_administrator) throw error.make(403, 'Insufficient permissions');
 
     if (req.body.__action === 'approve_deny_course_request') {
-      const id = req.body.request_id;
-      const user_id = res.locals.authn_user.user_id;
-      let action = req.body.approve_deny_action;
-
-      if (action === 'deny') {
-        action = 'denied';
-      } else {
-        throw new Error(`Unknown course request action "${action}"`);
-      }
-      await sqldb.queryOneRowAsync(sql.update_course_request, {
-        id,
-        user_id,
-        action,
-      });
-      res.redirect(req.originalUrl);
+      updateCourseRequest(req, res);
```

Comment:
This is an async function, you should await it for proper error handling.

Reply:
Now awaiting for this in commit [46a7046](https://github.com/PrairieLearn/PrairieLearn/pull/9238/commits/46a7046a6311dae7797f62cb4227fcbf1a3ead90).

Reply:
Now awaiting this in commit [a3022c1](https://github.com/PrairieLearn/PrairieLearn/pull/9238/commits/a3022c11669223d00a631b912db76424a485c694).